0x4db5362c

medium

# An error of overflow will be throned and cause unexpected behavior of function.

## Summary
An error of overflow could be throned when `D3UserQuota.sol#getUserQuota()` is called, which causes `getUserQuota()` return nothing except error.
## Vulnerability Detail
https://github.com/sherlock-audit/2023-06-dodo/blob/main/new-dodo-v3/contracts/DODOV3MM/D3Vault/periphery/D3UserQuota.sol#L89-L93
In the lines above, the calculation of unused quota doesn't compare two numbers before  subtraction, which will cause the function `getUserQuota()` throws an error.
## Impact
When `D3VaultFunding#userDeposit()` is called, the function could be reverted and no any specific error is throned out. But the design was intended to throw an error `Errors.EXCEED_QUOTA`

## Code Snippet
https://github.com/sherlock-audit/2023-06-dodo/blob/main/new-dodo-v3/contracts/DODOV3MM/D3Vault/periphery/D3UserQuota.sol#L89-L93

      if (isGlobalQuota[token]) {
          return (gloablQuota[token] - usedQuota) * 10 ** (_priceDecimal + tokenDecimals) / _tokenPrice;
      } else {
          return (calculateQuota(user) - usedQuota) * 10 ** (_priceDecimal + tokenDecimals) / _tokenPrice;
      }
## Tool used

Manual Review

## Recommendation
Compare the numbers before subtraction. Change the codes as below:

            if (isGlobalQuota[token]) {
                if (gloablQuota[token] > usedQuota) {
                    return (gloablQuota[token] - usedQuota) * 10 ** (_priceDecimal + tokenDecimals) / _tokenPrice;//@audit-info check implementation carefully
                } else {
                    return 0;
                }
            } else {
                uint256 quota = calculateQuota(user);
                if (quota > usedQuota) {
                    return (quota - usedQuota) * 10 ** (_priceDecimal + tokenDecimals) / _tokenPrice;//@audit-info check implementation carefully
                } else {
                    return 0;
                }
            }

